lms/djangoapps/commerce/tests/test_signals.py

```diff
@@ -154,14 +145,31 @@ def test_error_logging(self, mock_log_exception):
             self.assertTrue(mock_log_exception.called)
 
     @mock.patch('commerce.signals.send_refund_notification')
-    def test_notification(self, mock_send_notification):
+    def test_notification_when_approval_fails(self, mock_send_notification):
         """
-        Ensure the notification function is triggered when refunds are
-        initiated
+        Ensure the notification function is triggered when refunds are initiated, and cannot be automatically approved.
         """
-        with mock_create_refund(status=200, response=[1, 2, 3]):
-            self.send_signal()
-            self.assertTrue(mock_send_notification.called)
+        refund_id = 1
+        failed_refund_id = 2
+
+        with mock_create_refund(status=201, response=[refund_id, failed_refund_id]):
+            with mock_process_refund(refund_id):
+                with mock_process_refund(failed_refund_id, status=500):
+                    self.send_signal()
+                    self.assertTrue(mock_send_notification.called)
+                    mock_send_notification.assert_called_with(self.course_enrollment, [failed_refund_id])
+
+    @mock.patch('commerce.signals.send_refund_notification')
+    def test_no_notification_after_approval(self, mock_send_notification):
+        """
+        Ensure the notification function is triggered when refunds are initiated, and cannot be automatically approved.
+        """
+        refund_id = 1
+
+        with mock_create_refund(status=201, response=[refund_id]):
+            with mock_process_refund(refund_id):
```

Comment:
What about asserting that the correct action (`approve_payment_only`) was passed?